### PR TITLE
Save: Conversion to JSON

### PR DESCRIPTION
**Pull request by jmatholia**
Some CSS refactoring, fixes issue#386 and deletion of region elbows. Front end for save functionality. On save, SVG elements turned into JSON objects and sent to server as AJAX request.

### CODE REVIEW

File: hs/Css/DrawCss.hs [hunk 1]
```diff
@@ -40,53 +37,45 @@ mainCSS = "#main" ? do
...
-    left nil
+{- The SVG canvas and the grid background. -}
+canvasCSS = do
+    "#background" ? do
+        height100
+        width100
+        "background-image" -: "url(/static/res/grid.gif)"
+        "background-size" -: "128px"
+        opacity 0.3
+    "#mySVG" ? do
```
Comment (Ian-Stewart-Binks) [inline hs/Css/DrawCss.hs]: Hmm. We haven't been using camel case in IDs, and we haven't been using the `my` prefix. I realize that this name isn't new. I'll make a note of it to change in the future. There are a few names that do this.

File: hs/DrawSave.hs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+insertElements :: ([Path], [Shape], [Text]) -> IO ()
+insertElements (paths, shapes, texts) =
+    runSqlite dbStr $ do
+        mapM_ insert_ shapes
+        mapM_ insert_ paths
+        mapM_ insert_ texts
+
+
+-- | Type of each JSON entry in record syntax.
+data Person =
```
Comment (Ian-Stewart-Binks) [inline hs/DrawSave.hs]: Why are you pulling this into master?

**Comment by david-yz-liu** [comment]
These changes should be broken up into several different pull requests.

Also, remove commented out code and print statements. Also, remove notes to yourself. You should keep these in a separate location.

The indentation for save.js sometimes uses tabs. You need to configure your editor to use 4 spaces for tabs. 

Finally, you should be using the existing `createModalDiv` function (in common/modal.js) to create modals to maintain consistency.

File: public/js/draw/draw.js [hunk 3]
```diff
@@ -337,9 +338,8 @@ function unclickAll(e) {
 function finishRegion() {
     'use strict';
 
-    if (curPath !== null & curPath.elbows.length >= 3) {
+    if (mode === 'region-mode' & curPath !== null & curPath.elbows.length >= 3) {
```
Comment (Ian-Stewart-Binks) [inline public/js/draw/draw.js]: Why are you using the bitwise and operator?

File: public/js/draw/save.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
+
+function saveClicked() { // create the modal, correct way to do this?
+	'use strict';
+
+	var blackout = document.createElement('div');
+	blackout.setAttribute('id', 'save-modal-blackout');
+	var div = document.createElement('div');
+    div.setAttribute('id', 'save-modal');
```
Comment (Ian-Stewart-Binks) [inline public/js/draw/save.js]: Indentation issues.

File: public/js/draw/save.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+			method: 'get',
+			data: 'objs=' + JSON.stringify({'graph': {'gId': 100, 'title': savedAs}, 
+											'rects': rectAndText.rects, 
+											'texts': rectAndText.texts,
+											'paths': paths}),
+			success: function () { console.log('Ajax request sent!'); },
+			error: function (xhr, status, error) {
+						console.log('Error with Ajax request!');
+						console.log(status);
+						console.log(error);
```
Comment (Ian-Stewart-Binks) [inline public/js/draw/save.js]: Tabs...

**Pull request closed without merging**